index.go

```diff
@@ -372,8 +357,14 @@ func findR(cursor *bbolt.Cursor, sKey Key, matchers []matcher, fn iteratorFn) er
 					nKey := ComposeKey(sKey, newp)
 					err = findR(cursor, nKey, matchers[1:], fn)
 				} else {
-					// value found call iterator function
-					err = fn(cKey, entry)
+					// sub-bucket found, call iterator function
+					subBucket := cursor.Bucket().Bucket(cKey)
+					if subBucket != nil {
+						subCursor := subBucket.Cursor()
+						for k, _ := subCursor.Seek([]byte{}); k != nil; k, _ = subCursor.Next() {
```

Comment:
the cognitive and cyclomatic complexity here has just entered orbit... :'-)

Reply:
simplified the document retrieval, also noticed that the `cursor.seek` in the `else` wasn't even needed.